test/extensions/filters/http/csrf/csrf_filter_integration_test.cc

```diff
@@ -0,0 +1,200 @@
+#include "test/integration/http_protocol_integration.h"
+
+namespace Envoy {
+namespace {
+const std::string CSRF_ENABLED_CONFIG = R"EOF(
+name: envoy.csrf
+config:
+  filter_enabled:
+    default_value:
+      numerator: 100
+      denominator: HUNDRED
+  shadow_enabled:
+    default_value:
+      numerator: 100
+      denominator: HUNDRED
+)EOF";
+
+const std::string CSRF_FILTER_ENABLED_CONFIG = R"EOF(
+name: envoy.csrf
+config:
+  filter_enabled:
+    default_value:
+      numerator: 100
+      denominator: HUNDRED
+)EOF";
+
+const std::string CSRF_SHADOW_ENABLED_CONFIG = R"EOF(
+name: envoy.csrf
+config:
+  filter_enabled:
+    default_value:
+      numerator: 0
+      denominator: HUNDRED
+  shadow_enabled:
+    default_value:
+      numerator: 100
+      denominator: HUNDRED
+)EOF";
+
+const std::string CSRF_DISABLED_CONFIG = R"EOF(
+name: envoy.csrf
+config:
+  filter_enabled:
+    default_value:
+      numerator: 0
+      denominator: HUNDRED
+)EOF";
+
+class CsrfFilterIntegrationTest : public HttpProtocolIntegrationTest {
+protected:
+  void testNormalRequest(Http::TestHeaderMapImpl&& request_headers,
+                         const char* expected_response_code) {
```

Comment:
pass a absl::string_view, here and below. Avoid use of .c_str() for the comparison too as that's going away.